flexget/plugins/list/subtitle_list.py

```diff
@@ -0,0 +1,261 @@
+
+
+class SubtitleList(MutableSet):
+    schema = {
+        'type': 'object',
+        'properties': {
+            'list': {'type': 'string'},
+            'languages': {'type': 'array', 'items': {'type': 'string'}, 'minItems': 1},
+            'check_subtitles': {'type': 'boolean', 'default': True},
+            'remove_after': {'type': 'string', 'format': 'interval'}
+        },
+        'required': ['list'],
+        'additionalProperties': False
+    }
+
+    def _db_list(self, session):
+        return session.query(SubtitleListList).filter(SubtitleListList.name == self.config['list']).first()
+
+    def _from_iterable(self, it):
+        # TODO: is this the right answer? the returned object won't have our custom __contains__ logic
+        return set(it)
+
+    @with_session
+    def __init__(self, config, session=None):
+        self.config = config
+        db_list = self._db_list(session)
+        if not db_list:
+            session.add(SubtitleListList(name=self.config['list']))
+
+    @with_session
+    def __iter__(self, session=None):
+        return iter([file.to_entry() for file in self._db_list(session).files])
+
+    @with_session
+    def __len__(self, session=None):
+        return self._db_list(session).files.count()
+
+    @with_session
+    def add(self, entry, session=None):
+        # Check if this is already in the list, refresh info if so
+        db_list = self._db_list(session=session)
+        db_file = self._find_entry(entry, session=session)
+        # Just delete and re-create to refresh
+        if db_file:
+            session.delete(db_file)
+        db_file = SubtitleListFile()
+        db_file.title = entry['title']
+        db_file.location = entry['location']
+        db_file.languages = []
+        db_file.remove_after = self.config.get('remove_after')
+        languages = set()
+        for subtitle_language in self.config.get('languages', []):
+            normalized_language = normalize_language(subtitle_language)
+            language = self._find_language(normalized_language)
+            if not language:
+                language = SubtitleListLanguage(language=normalized_language)
+            languages.add(language)
+        db_file.languages.extend(list(languages))
+        log.debug('adding entry %s', entry)
+        db_list.files.append(db_file)
+        session.commit()
+        return db_file.to_entry()
+
+    @with_session
+    def discard(self, entry, session=None):
+        db_file = self._find_entry(entry, session=session)
+        if db_file:
+            log.debug('deleting file %s', db_file)
+            session.delete(db_file)
+
+    def __contains__(self, entry):
+        return self._find_entry(entry) is not None
+
+    @with_session
+    def _find_entry(self, entry, session=None):
+        """Finds `SubtitleListFile` corresponding to this entry, if it exists."""
+        res = self._db_list(session).files.filter(SubtitleListFile.location == entry.get('location', '')).first()
```

Comment:
I think it's fine `SubtitleListList.files` is a dynamic relationship, which will be a query on `SubtitleListFile`s that are children of the `SubtitleListList`